hooks/RMC.js

```diff
@@ -39,25 +39,23 @@ values:
 module.exports = function (input) {
   const { id, sentence, parts, tags } = input
 
-  let latitude = -1
-  let longitude = -1
-  let speed = 0.0
-  let track = 0.0
-  let variation = 0.0
+  let latitude = null
+  let longitude = null
+  let speed = null
+  let track = null
+  let variation = null
 
   const timestamp = utils.timestamp(parts[0], parts[8])
   const age = moment.tz(timestamp, 'UTC').unix()
 
-  latitude = utils.coordinate(parts[2], parts[3])
-  longitude = utils.coordinate(parts[4], parts[5])
+  latitude = !isNaN(parseFloat(parts[2])) && isFinite(parts[2]) && "NS".includes(parts[3]) ? utils.coordinate(parts[2], parts[3]) : null
```

Comment:
`isFinite(NaN)` is false, so `isNan` is not really needed and adds an extra parseFloat to everything?

Reply:
Agreed, just wasn't familiar with the proper/preferred way to test for null with Chai.    Latest commit has two test sentences which check for valid & null entries with SOC/COG & variation.

Reply:
I think you read my comment out of context? I meant that we don't need both `!isNaN` and `isFinite` in the ternary's condition.

Reply:
Since javascript coerces empty strings to 0 and !isNaN('') returns true it wasn't straight forward with a single test.  My original though for testing was:   
parseFloat(parts[2])).trim().length > 0 && !isNaN(parseFloat(parts[2]))
I looked around for the best/preferred way to determine if a string was a valid number but I see what I found has issues.
I can update with the test above or if you have a preferred way please let me know.  Thanks.

Reply:
Err:  parts[2]).trim().length > 0 && !isNaN(parts[2])
Seems so simple  ;)

Reply:
I went ahead and committed the data validity check shown above.  Hopefully that looks good.  If not happy to replace with a preferred method & wrap up this small but important change for RMC.